butterknife-compiler/src/main/java/butterknife/compiler/ButterKnifeProcessor.java

```diff
@@ -1143,17 +1147,19 @@ private String getPackageName(TypeElement type) {
     return elementUtils.getPackageOf(type).getQualifiedName().toString();
   }
 
-  private static boolean hasAnnotationWithName(Element element, String simpleName) {
+  private static boolean hasAnnotationWithNames(Element element, String... simpleNames) {
     for (AnnotationMirror mirror : element.getAnnotationMirrors()) {
       String annotationName = mirror.getAnnotationType().asElement().getSimpleName().toString();
-      if (simpleName.equals(annotationName)) {
-        return true;
+      for (String simpleName : simpleNames) {
+        if (simpleName.equals(annotationName)) {
+          return true;
+        }
       }
     }
     return false;
   }
 
   private static boolean isRequiredBinding(Element element) {
-    return !hasAnnotationWithName(element, NULLABLE_ANNOTATION_NAME);
+    return !hasAnnotationWithNames(element, NULLABLE_ANNOTATION_NAME, OPTIONAL_ANNOTATION_NAME);
```

Comment:
I thought separate methods thinking there wouldn't be too much else that had to change. If it truly is a lot, I can take a closer look.